packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -275,6 +275,9 @@ class FlutterDevice {
             existingDds = true;
           }
         } on Exception catch (e, st) {
+          if (e is ToolExit) {
```

Comment:
Would it be more idiomatic to handle this in an `on ToolExit` block that precedes the `on Exception` block?

Reply:
Yeah, it would be. Clearly still half asleep.